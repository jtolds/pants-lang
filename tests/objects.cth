obj = construct { |o|
  o.field1 = 3
  o.field2 = 4
}

print obj
print obj.field1
print obj.field2
obj.field1 = 5
print obj.field1
print "should make it here"
obj.field3 = 6 # should fail!
print "shouldn't make it here"
