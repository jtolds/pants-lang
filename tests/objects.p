# PANTS OPTIONS: --skip-prelude

constructor = {|func|
  {|:(largs); :(rargs), ::(kwargs)|
    obj = new_object.
    func(:(largs); obj, :(rargs), ::(kwargs))
    seal_object obj
    obj
  }
}
# TODO: actually clear this out of the scope or something
new_object = null
seal_object = null

obj = (constructor { |o|
  o.field1 = 3
  o.field2 = 4
})()

println obj
println obj.field1
println obj.field2
obj.field1 = 5
println obj.field1
println "should make it here"
obj.field3 = 6 # should fail!
println "shouldn't make it here"
