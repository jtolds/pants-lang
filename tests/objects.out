<TODO: object>
3
4
5
should make it here
Exception thrown!
=> object u_obj sealed!

return code: 1
